### ISSUE TITLE
Process for component suggestions

### ISSUE THREAD

**Issue by emplums** [primer/components#767]
We need some docs written up on how to propose a component/what that process is. Let's add these to primer.style/components as well as the Readme.

These docs should include:
- Look into what the new component proposal is for Primer CSS and try to keep some consistency
- Decision tree for whether or not a component should live in Primer Components
- Should folks get an OK from the core team before starting work so they don't waste their time on something that ultimately we might not add? What would be the process for this? Let's document that
- How to open a PR for a new component/what kinds of things should be in the PR (maybe this should just be a new PR template?)
- Should we have an "experimental components" repo where we can host beta components before adding them to the Primer Components package? Would something like this get used? I'm thinking something sort of like a staging/beta package - not every component would get added to this package first, just ones that we want to play with in beta for a bit.

**Comment by yaili** [comment]
I'd love to help, if needed. 🤗😍

For Azure DevOps we had:
- [Component lifecycle](https://azdevinternal.azureedge.net/getting-started/component-lifecycle)
- [Ways to contribute](https://azdevinternal.azureedge.net/getting-started/contributing)

This was early days, and we didn't get a chance to test it too often, but was a start and I think people found it useful.

I think it's also important to be clear how new components will be reviewed. Like, do we have coding standards and other checklists that would make those things clear? Example [here](https://vanillaframework.io/contribute) it links to a few of those docs and that's all documented.

**Comment by emplums** [comment]
This was done in https://github.com/primer/components/pull/815!
